### PR TITLE
Add filepath glob support to --security-opt unmask

### PR DESCRIPTION
**Pull request by rhatdan**
Want to allow users to specify --security-opt unmask=/proc/*.
This allows us to run podman within podman more securely, then
specifing umask=all, also gives the user more flexibilty.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@rhatdan still fighting the CI test system

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL

**Review by ashley-cui — COMMENTED** [review]
LGTM

File: contrib/podmanimage/stable/containers.conf [hunk 1]
```diff
@@ -5,6 +5,7 @@ ipcns="host"
 utsns="host"
 cgroupns="host"
 cgroups="disabled"
+log_driver = "k8s_file"
```
Comment (vrothberg) [inline contrib/podmanimage/stable/containers.conf:8]: Is this change intentional?

File: pkg/util/utils.go [hunk 2]
```diff
@@ -76,6 +76,24 @@ func ParseRegistryCreds(creds string) (*types.DockerAuthConfig, error) {
 	}, nil
 }
 
+func ShouldMask(mask string, unmask []string) bool {
```
Comment (vrothberg) [inline pkg/util/utils.go:79]: Since there's only one callers, can we move it into `pkg/specgen/generate`?

Comment (vrothberg) [reply pkg/util/utils.go:79]: Code comments and unit tests would be nice.

File: contrib/podmanimage/stable/containers.conf (hunk 1 above)
Comment (rhatdan) [reply contrib/podmanimage/stable/containers.conf:8]: Yes it is a separate PR, Could add it as a fully separate from this, but I figured I would sneak it in while I was thinking about it.

**Review by giuseppe — APPROVED** [review]
LGTM

**Comment by TomSweeneyRedHat** [comment]
LGTM
